client/engine/store/mockstore.go

```diff
@@ -0,0 +1,53 @@
+package store
+
+import (
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+type MockStore struct {
+	objectives map[protocols.ObjectiveId]protocols.Objective
+
+	key []byte // the signing key of the store's engine
+}
+
+func NewMockStore(key []byte) Store {
+	ms := MockStore{}
+	ms.key = key
+	ms.objectives = make(map[protocols.ObjectiveId]protocols.Objective)
+
+	return ms
+}
+
+func (ms MockStore) GetChannelSecretKey() *[]byte {
+	return &ms.key
+}
+
+func (ms MockStore) GetObjectiveById(id protocols.ObjectiveId) (obj protocols.Objective, ok bool) {
+	// todo: locking
```

Comment:
locking is potentially most important with respect to the ledger channel between Irene and Irma. Consider:

```
Alice             Bob 
    \            / 
     Irene - Irma 
    /            \
Andy              Beth
```

If (Alice and Bob) and (Andy and Beth) simultaneously attempt to open virtual channels, this will create multiple funding objectives for each of Irene, Irma, which reference the same I-I ledger channel and can update the state of that channel.

We don't want race conditions here, where changes from one objective overwrite those from the other, or prevent those from the other from taking place.